src/approaches/nsrt_rl_approach.py

```diff
@@ -61,126 +60,134 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
         # skeleton. The teacher will collect a trajectory on the training task
         # using this policy.
         requests = []
+        self._requests = []
         for i in range(len(self._train_tasks)):
             task = self._train_tasks[i]
             try:
                 _act_policy = self.solve(task, CFG.timeout)
-                self._train_task_to_option_plan[i] = self._last_plan
+                self._requests.append((i, self._last_plan))
             except (ApproachTimeout, ApproachFailure) as e:
                 partial_refinements = e.info.get("partial_refinements")
                 assert partial_refinements is not None
                 _, plan = max(partial_refinements, key=lambda x: len(x[1]))
                 _act_policy = utils.option_plan_to_policy(plan)
-                self._train_task_to_option_plan[i] = plan
+                self._requests.append((i, plan))
             request = InteractionRequest(train_task_idx=i,
                                          act_policy=_act_policy,
                                          query_policy=lambda s: None,
                                          termination_function=task.goal_holds)
             requests.append(request)
         return requests
 
+    def _get_experience_from_result(
+        self, idx: int, result: InteractionResult
+    ) -> Dict[ParameterizedOption, List[List[Tuple[State, Array, Action, int,
+                                                   State]]]]:
+        option_to_data: Dict[ParameterizedOption,
+                             List[List[Tuple[State, Array, Action, int,
+                                             State]]]] = {}
+        train_task_idx, plan = self._requests[idx]
+        traj = LowLevelTrajectory(result.states,
+                                  result.actions,
+                                  _is_demo=False,
+                                  _train_task_idx=train_task_idx)
+        curr_option_idx = 0
+        curr_option = plan[curr_option_idx]
+        parent_option = cast(_LearnedNeuralParameterizedOption,
+                             curr_option.parent)
+        if parent_option not in option_to_data:
+            option_to_data[parent_option] = []
+        experience = []
+
+        # Loop through the trajectory and compute the experience tuples for
+        # each _Option in the plan.
+        j = 0
+        for state, next_state in zip(traj.states[:-1], traj.states[1:]):
```

Comment:
you can do instead:

```for state, action, next_state in zip(traj.states[:-1], traj.actions, traj.states[1:]):```

and then get rid of `j`

Reply:
j is used in line 142.